torch_xla/distributed/spmd/xla_sharding.py

```diff
@@ -524,7 +530,7 @@ def mark_sharding(
 
         mesh (Mesh): describes the logical XLA device topology and the underlying device IDs.
 
-        partition_spec (Tuple[Tuple, int, str, None]): A tuple of device_mesh dimension index or
+        partition_spec (PartitionSpec): A tuple of device_mesh dimension index or
             `None`. Each index is an int, str if the mesh axis is named, or tuple of int or str.
```

Comment:
This makes it sound like there are two options: 1 each index is an int, str tuple or 2 each index is an int, str tuple. But both options are the same. Am I misunderstanding?

Reply:
Each element can be:

- int (refer to mesh axis by index)
- str (refer to mesh axis by name)
- a tuple of the above

In the third case, the corresponding dimension of the tensor will be sharded over multiple mesh axes.

..now that i wrote this out i might as well rewrite this whole confusing section too..

Reply:
Ah, ok! Great job explaining this!